src/core/tests/preprocess.cpp

```diff
@@ -534,6 +534,31 @@ TEST(pre_post_process, reuse_model_layout_no_tensor_info) {
     EXPECT_EQ(f->get_parameters().front()->get_layout(), "NC??");
 }
 
+TEST(pre_post_process, set_layout_out_of_bounds) {
+    auto shape = PartialShape{1, 2};
+    std::stringstream shape_str;
+    shape_str << shape;
+    auto f = create_simple_function(element::f32, shape);
+    Layout from{"NHWC"};
+    Layout to{"NCHW"};
+    // TODO: replace with EXPECT_THAT after upgrade gtest to v1.11
```

Comment:
BTW, you can try to migrate to v1.11 using this state https://github.com/openvinotoolkit/googletest/tree/release-1.11.0-openvino